items:
- uid: System.Runtime.InteropServices.COMException
  commentId: T:System.Runtime.InteropServices.COMException
  id: COMException
  parent: System.Runtime.InteropServices
  children:
  - System.Runtime.InteropServices.COMException.#ctor
  - System.Runtime.InteropServices.COMException.#ctor(System.String)
  - System.Runtime.InteropServices.COMException.#ctor(System.String,System.Exception)
  - System.Runtime.InteropServices.COMException.#ctor(System.String,System.Int32)
  langs:
  - csharp
  - vb
  name: COMException
  fullName: System.Runtime.InteropServices.COMException
  type: Class
  source:
    remote:
      path: src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: COMException
    path: src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs
    startLine: 128
  assemblies:
  - System.Runtime.InteropServices
  namespace: System.Runtime.InteropServices
  summary: "\n<p>The exception that is thrown when an unrecognized HRESULT is returned from a COM method call.</p>\n"
  remarks: "\n<p>The common language runtime transforms well-known HRESULTs to .NET Framework exceptions, enabling COM objects to return meaningful error information to managed clients. The HRESULT to exception mapping also works in the other direction by returning specific HRESULTs to unmanaged clients. For mapping details, see How to: Map HRESULTs and Exceptions.</p>\n<p>When the runtime encounters an unfamiliar HRESULT (an HRESULT that lacks a specific, corresponding exception), it throws an instance of the <xref href=\"System.Runtime.InteropServices.COMException\" data-throw-if-not-resolved=\"false\"></xref> class. This all-purpose exception exposes the same members as any exception, and inherits a public <xref href=\"System.Runtime.InteropServices.ExternalException.ErrorCode\" data-throw-if-not-resolved=\"false\"></xref> property that contains the HRESULT returned by the callee. If an error message is available to the runtime (obtained from the <see href=\"https://msdn.microsoft.com/library/windows/desktop/ms723041(v=vs.85).aspx\">IErrorInfo</see> interface or the <strong>Err</strong> object in Visual Basic, or in some cases from the operating system), the message is returned to the caller. However, if the COM component developer fails to include an error message, the runtime returns the eight-digit HRESULT in place of a message string. Having an HRESULT allows the caller to determine the cause of the generic exception.</p>\n"
  syntax:
    content: 'public class COMException : Exception'
    content.vb: >-
      Public Class COMException
          Inherits Exception
  seealso:
  - type: System.Exception
    commentId: T:System.Exception
  see:
  - type: System.Runtime.InteropServices.ExternalException.ErrorCode
    commentId: P:System.Runtime.InteropServices.ExternalException.ErrorCode
  inheritance:
  - System.Object
  modifiers.csharp:
  - public
  - class
  modifiers.vb:
  - Public
  - Class
- uid: System.Runtime.InteropServices.COMException.#ctor
  commentId: M:System.Runtime.InteropServices.COMException.#ctor
  id: '#ctor'
  parent: System.Runtime.InteropServices.COMException
  langs:
  - csharp
  - vb
  name: COMException()
  fullName: System.Runtime.InteropServices.COMException.COMException()
  type: Constructor
  source:
    remote:
      path: src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: .ctor
    path: src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs
    startLine: 130
  assemblies:
  - System.Runtime.InteropServices
  namespace: System.Runtime.InteropServices
  summary: "\n<p>Initializes a new instance of the <xref href=\"System.Runtime.InteropServices.COMException\" data-throw-if-not-resolved=\"false\"></xref> class with default values.</p>\n"
  remarks: "\n<p>\n<xref href=\"System.Runtime.InteropServices.COMException\" data-throw-if-not-resolved=\"false\"></xref> inherits from <xref href=\"System.Runtime.InteropServices.ExternalException\" data-throw-if-not-resolved=\"false\"></xref>. The following table shows how this constructor sets the properties of the <xref href=\"System.Exception\" data-throw-if-not-resolved=\"false\"></xref> object.</p>\n<p>Property </p>\n<p>Value </p>\n<table><tbody><tr><td>\n<p>\n<xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>\n<strong>null</strong>. </p>\n</td></tr><tr><td>\n<p>\n<xref href=\"System.Exception.Message\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>A localized error message string. </p>\n</td></tr></tbody></table>\n"
  syntax:
    content: public COMException()
    content.vb: Public Sub New
  modifiers.csharp:
  - public
  modifiers.vb:
  - Public
- uid: System.Runtime.InteropServices.COMException.#ctor(System.String)
  commentId: M:System.Runtime.InteropServices.COMException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.Runtime.InteropServices.COMException
  langs:
  - csharp
  - vb
  name: COMException(String)
  fullName: System.Runtime.InteropServices.COMException.COMException(System.String)
  type: Constructor
  source:
    remote:
      path: src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: .ctor
    path: src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs
    startLine: 131
  assemblies:
  - System.Runtime.InteropServices
  namespace: System.Runtime.InteropServices
  summary: "\n<p>Initializes a new instance of the <xref href=\"System.Runtime.InteropServices.COMException\" data-throw-if-not-resolved=\"false\"></xref> class with a specified message.</p>\n"
  remarks: "\n<p>\n<xref href=\"System.Runtime.InteropServices.COMException\" data-throw-if-not-resolved=\"false\"></xref> inherits from <xref href=\"System.Runtime.InteropServices.ExternalException\" data-throw-if-not-resolved=\"false\"></xref>. The following table shows how this constructor sets the properties of the <xref href=\"System.Exception\" data-throw-if-not-resolved=\"false\"></xref> object.</p>\n<p>Property </p>\n<p>Value </p>\n<table><tbody><tr><td>\n<p>\n<xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>\n<strong>null</strong>\n</p>\n</td></tr><tr><td>\n<p>\n<xref href=\"System.Exception.Message\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>\n<em>message</em>\n</p>\n</td></tr></tbody></table>\n"
  syntax:
    content: public COMException(string message)
    content.vb: Public Sub New(message As String)
    parameters:
    - id: message
      type: System.String
      description: "\n<p>The message that indicates the reason for the exception. </p>\n"
  modifiers.csharp:
  - public
  modifiers.vb:
  - Public
- uid: System.Runtime.InteropServices.COMException.#ctor(System.String,System.Exception)
  commentId: M:System.Runtime.InteropServices.COMException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.Runtime.InteropServices.COMException
  langs:
  - csharp
  - vb
  name: COMException(String, Exception)
  fullName: System.Runtime.InteropServices.COMException.COMException(System.String, System.Exception)
  type: Constructor
  source:
    remote:
      path: src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: .ctor
    path: src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs
    startLine: 132
  assemblies:
  - System.Runtime.InteropServices
  namespace: System.Runtime.InteropServices
  summary: "\n<p>Initializes a new instance of the <xref href=\"System.Runtime.InteropServices.COMException\" data-throw-if-not-resolved=\"false\"></xref> class with a specified error message and a reference to the inner exception that is the cause of this exception.</p>\n"
  remarks: "\n<p>An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref> property. The <xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref> property returns the same value that is passed into the constructor, or <strong>null</strong> if the <xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref> property does not supply the inner exception value to the constructor.</p>\n<p>The following table shows how this constructor sets the properties of the <xref href=\"System.Exception\" data-throw-if-not-resolved=\"false\"></xref> object.</p>\n<p>Property </p>\n<p>Value </p>\n<table><tbody><tr><td>\n<p>\n<xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>The inner exception reference. </p>\n</td></tr><tr><td>\n<p>\n<xref href=\"System.Exception.Message\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>The error message string. </p>\n</td></tr></tbody></table>\n"
  syntax:
    content: public COMException(string message, Exception inner)
    content.vb: Public Sub New(message As String, inner As Exception)
    parameters:
    - id: message
      type: System.String
      description: "\n<p>The error message that explains the reason for the exception. </p>\n"
    - id: inner
      type: System.Exception
      description: "\n<p>The exception that is the cause of the current exception. If the <em>inner</em> parameter is not <strong>null</strong>, the current exception is raised in a <strong>catch</strong> block that handles the inner exception. </p>\n"
  seealso:
  - type: System.Exception
    commentId: T:System.Exception
  modifiers.csharp:
  - public
  modifiers.vb:
  - Public
- uid: System.Runtime.InteropServices.COMException.#ctor(System.String,System.Int32)
  commentId: M:System.Runtime.InteropServices.COMException.#ctor(System.String,System.Int32)
  id: '#ctor(System.String,System.Int32)'
  parent: System.Runtime.InteropServices.COMException
  langs:
  - csharp
  - vb
  name: COMException(String, Int32)
  fullName: System.Runtime.InteropServices.COMException.COMException(System.String, System.Int32)
  type: Constructor
  source:
    remote:
      path: src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: .ctor
    path: src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs
    startLine: 133
  assemblies:
  - System.Runtime.InteropServices
  namespace: System.Runtime.InteropServices
  summary: "\n<p>Initializes a new instance of the <xref href=\"System.Runtime.InteropServices.COMException\" data-throw-if-not-resolved=\"false\"></xref> class with a specified message and error code.</p>\n"
  remarks: "\n<p>This <xref href=\"System.Runtime.InteropServices.COMException\" data-throw-if-not-resolved=\"false\"></xref> constructor sets the base message with the error code.</p>\n<p>\n<xref href=\"System.Runtime.InteropServices.COMException\" data-throw-if-not-resolved=\"false\"></xref> inherits from <xref href=\"System.Runtime.InteropServices.ExternalException\" data-throw-if-not-resolved=\"false\"></xref>. The following table shows how this constructor sets the properties of the <xref href=\"System.Exception\" data-throw-if-not-resolved=\"false\"></xref> object.</p>\n<p>Property </p>\n<p>Value </p>\n<table><tbody><tr><td>\n<p>\n<xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>\n<strong>null</strong>\n</p>\n</td></tr><tr><td>\n<p>\n<xref href=\"System.Exception.Message\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>\n<em>message</em>\n</p>\n</td></tr></tbody></table>\n"
  syntax:
    content: public COMException(string message, int errorCode)
    content.vb: Public Sub New(message As String, errorCode As Integer)
    parameters:
    - id: message
      type: System.String
      description: "\n<p>The message that indicates the reason the exception occurred. </p>\n"
    - id: errorCode
      type: System.Int32
      description: "\n<p>The error code (HRESULT) value associated with this exception. </p>\n"
  modifiers.csharp:
  - public
  modifiers.vb:
  - Public
references:
- uid: System.Runtime.InteropServices
  isExternal: false
  name: System.Runtime.InteropServices
  fullName: System.Runtime.InteropServices
- uid: System.Object
  parent: System
  isExternal: false
  name: Object
  fullName: System.Object
- uid: System
  isExternal: false
  name: System
  fullName: System
- uid: System.String
  parent: System
  isExternal: false
  name: String
  fullName: System.String
- uid: System.Exception
  parent: System
  isExternal: false
  name: Exception
  fullName: System.Exception
- uid: System.Int32
  parent: System
  isExternal: false
  name: Int32
  fullName: System.Int32
